mitodl/ocw-studio
# Add ckeditor5 math plugin

#### Pre-Flight checklist


#### What are the relevant tickets?
#1482 ; This PR is a rebased/updated version of #1512 

#### What's this PR do?
This PR adds support for LaTeX / mathjax editing via a community-built [ckeditor5 math plugin](https://github.com/isaul32/ckeditor5-math).

- Adds Mathjax to render math; version 3 like OCW uses 
- Adds the aforementioned ckeditor5 math plugin
- Adds a MathSyntax plugin for MD <--> HTML conversion. 

#### How should this be manually tested?
1. Open a course page for editing
2. Use the new "f(x)" button to enter math. Math can be entered in either inline or display (similar to html block) mode.
3. Try entering lots of math, including stuff like inequalities and other potential HTML special characters. The math must be valid latex.

    Here's some sample math to try:
    ```
    \frac{1}{2}
    \sqrt{x^2 + y^2}
    a \leq b < c
    a \geq b > c
    \sin{\left(\frac{\pi}{3}\sqrt[3]{x+y}\right)}
    \begin{pmatrix} a & b \\ c & d \end{pmatrix}
    
    chemistry stuff:
    \ce{H20}
    \ce{NH4+}
    \ce{CH3COOH + OH-} \rightleftharpoons \ce{CH3COO- + H2O}
    ```
4. Check that the resulting markdown actually renders in OCW: Get the resulting markdown in django admin and load it up in any ocw-hugo-themes course.

#### Screenshots (if appropriate)
<img width="1324" alt="Screen Shot 2022-10-11 at 12 21 27 PM" src="https://user-images.githubusercontent.com/9010790/195146469-d948584f-2846-41ee-8894-bbbe9342bf4d.png">

<img width="872" alt="Screen Shot 2022-10-11 at 11 46 28 AM" src="https://user-images.githubusercontent.com/9010790/195146356-422c31ef-74d9-45bb-89a0-dfca3a2d288a.png">

### Additional Context
This also allows entering math in ocw-www. Currently, www does not support mathjax, but https://github.com/mitodl/ocw-hugo-themes/pull/906 allows it (and switches our hugo theme to only including mathjax if it is used.)
